src/sentry/eventstream/snuba.py

```diff
@@ -118,6 +119,12 @@ def insert(
         skip_consume: bool = False,
         **kwargs: Any,
     ) -> None:
+        if isinstance(event, GroupEvent):
+            logger.error(
+                "`GroupEvent` passed to `EventStream.insert`. Only `Event` is allowed here.",
+                exc_info=True,
+            )
+            return
```

Comment:
Did you mean to put this on SnubaEventStream instead of SnubaProtocolEventStream (which is the base class that isn't used directly? For Kafka backend (which also inherits from this, this means we are checking this twice currently.

Reply:
Just picking up this pr from a while ago, I forgot that this base class is used by both. I'll just leave it in the one place